backend/qemu.pm

```diff
@@ -186,14 +186,15 @@ sub do_extract_assets {
     my $hdd_num = $args->{hdd_num};
     my $name    = $args->{name};
     my $img_dir = $args->{dir};
-    my $format  = $args->{format};
+    $name =~ /\.([[:alnum:]]+)$/;
+    my $format = $1;
```

Comment:
this is now very confusing. You strip the format off $name - and then also remove the code that appends it